### PR TITLE
RHODS-12818: Updating package versions.

### PR DESCRIPTION
**Pull request by grainnejenningsRH**
Updating package versions to align with latest.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: modules/options-for-notebook-server-environments.adoc [hunk 1]
```diff
@@ -278,7 +278,7 @@ a| * Python 3.8
 * Pandas 1.5
 * Scikit-learn 1.2
 * SciPy 1.10
-* PyTorch 2.0
+* PyTorch 2.0.1
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:281]: We only mention Major and Minor versions of the tools in this list (x.y). So please revert this line back to:

```suggestion
* PyTorch 2.0
```

File: modules/options-for-notebook-server-environments.adoc [hunk 2]
```diff
@@ -87,7 +87,7 @@ a| * CUDA 11.8
 * SciPy 1.11
 * Elyra 3.15
 * PyMongo 4.5 
-* Pyobc 4.0 
+* Pyodbc 4.0
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:90]: Very nice catch! Thanks :slightly_smiling_face:

**Review by jstourac — CHANGES_REQUESTED** [review]
Thank you very much! I put one comment in the code for update.

Also, I think it is worth to mention version of JupyterLab for the HabanaAI image. Could you please add this information on appropriate [line](https://github.com/opendatahub-io/opendatahub-documentation/pull/88/files#diff-ee0f169abaa07a514e305ab12b8e6d4114d7111f7d6813f545b64a0e3ad453aeR271)?

It should be like this after the change:

```
a| * Python 3.8
* Habana 1.10
* JupyterLab 3.5
* TensorFlow 2.12
```

Otherwise the changes look good to me :slightly_smiling_face:

File: modules/options-for-notebook-server-environments.adoc (hunk 1 above)
Comment (grainnejenningsRH) [reply modules/options-for-notebook-server-environments.adoc]: Thanks Jan! Have updated.

**Review by jstourac — APPROVED** [review]
LGTM, thank you!

**Pull request merged**